analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/MethodReference.cs

```diff
@@ -0,0 +1,33 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2024 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.SymbolicExecution.Roslyn.OperationProcessors;
+
+internal sealed class MethodReference : SimpleProcessor<IMethodReferenceOperationWrapper>
+{
+    protected override IMethodReferenceOperationWrapper Convert(IOperation operation) =>
+        IMethodReferenceOperationWrapper.FromOperation(operation);
+
+    protected override ProgramState Process(SymbolicContext context, IMethodReferenceOperationWrapper operation) =>
+        CollectionTracker.MethodReferenceConstraint(operation) is { } constraint
+        && operation.Instance.TrackedSymbol(context.State) is { } symbol
+            ? context.State.SetSymbolConstraint(symbol, constraint)
```

Comment:
before, we setted both the symbol and the operation of the instance.
I do not see a reason to set the _instance_ operation of the `MethodReference`, since I don't think it can be re-used.